### ISSUE TITLE
[bug] Unable to bundle external binaries with the app (macOS)

### ISSUE THREAD

**Issue by pronebird** [tauri-apps/tauri#9422]
### Describe the bug

Hi,

It seems that `externalBin` setting does not work as expected on Mac. 

This is my configuration:

```
    "externalBin": [
      "../../target/universal-apple-darwin/release/binary-a",
      "../../target/universal-apple-darwin/release/binary-b",
      "../../target/universal-apple-darwin/release/binary-c"
    ],
```

So then I go ahead and rebuild my external binaries and copy them into `target/universal-apple-darwin/release` with each binary having an arch suffix, i.e:

```
$ ls -1 target/universal-apple-darwin/release/

My App
bundle
binary-a-aarch64-apple-darwin
binary-b-aarch64-apple-darwin
binary-b-x86_64-apple-darwin
binary-c-aarch64-apple-darwin
binary-c-x86_64-apple-darwin
binary-a-x86_64-apple-darwin
```

Yet somehow tauri build fails with:

> Error failed to bundle project: Failed to copy external binaries

If this feature is broken, is there any way to tell tauri to codesign additional binaries, for example I can easily embed them into `Resources/` directory or should I implement code signing myself?

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 14.2.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.79.0-nightly (4fd4797c2 2024-04-03)
    ✔ cargo: 1.79.0-nightly (0637083df 2024-04-02)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: nightly-x86_64-apple-darwin (default)
    - node: 21.7.1
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.14 (no manifest)
    - tauri-build [RUST]: 2.0.0-beta.11 (no manifest)
    - wry [RUST]: 0.39.0 (no manifest)
    - tao [RUST]: 0.27.0 (no manifest)
    - @tauri-apps/api [NPM]: 2.0.0-beta.6
    - @tauri-apps/cli [NPM]: 2.0.0-beta.11

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

**Comment by FabianLars** [comment]
Can you try again with the --verbose flag and see if it gives you a more helpful error message? iirc the cli/bundler don't handle extra binaries for universal targets yet so maybe that's the reason here and no misconfig on your side.

**Comment by pronebird** [comment]
Unfortunately it doesn't print anything but:

> Error [tauri_cli_node] failed to bundle project: Failed to copy external binaries

For the record I am running CLI with:

```sh
npm run tauri build -- --verbose --target universal-apple-darwin
```

The whole thing breaks before the `<MY_APP.app>/MacOS` is created but after resources are copied into `<MY_APP.app>/Resources`

It seems that it just doesn't know how to merge binaries together. It would be great to run `lipo` per binary:

```
lipo -create "<MY_BIN>-x86_64-apple-darwin" "<MY_BIN>-aarch64-apple-darwin" -output "<MY_APP.app>/MacOS/<MY_BIN>"
```

This would offload my build.sh script by quite a bit as I have to run `lipo` manually before `tauri build`

**Comment by lbenlake** [comment]
Any fix for this issue? I am also experiencing it.

**Comment by lbenlake** [comment]
Does this have anything to do with this one? https://github.com/tauri-apps/tauri/issues/8152

**Comment by FabianLars** [comment]
Yes, it's the same issue. I'm honestly not quite sure how we should fix this, i think just documenting that a universal tauri app needs universal sidecars (via lipo) because that feels more in line with the bundler's current behavior elsewhere 🤔 

cc @lucasfernog

**Comment by pronebird** [comment]
@FabianLars it just hit me, does bundler look for `<BIN_NAME>-universal-apple-darwin` when copying the file?

**Comment by FabianLars** [comment]
i think it does, yes. But iirc at runtime it tries to run the sidecar for the actual arch that's running instead of looking for the universal one.

(could also be the other way around, but at least one part is broken)
